### PR TITLE
334 email verification on flutter

### LINKED ISSUE #334: email verification on flutter

### PR DESCRIPTION
**Pull request by VertexToEdge**
close #334 

https://user-images.githubusercontent.com/37045096/113993883-dbc81400-988f-11eb-88f7-c3029e6da837.mp4

University Email verification 

https://user-images.githubusercontent.com/37045096/113993465-76742300-988f-11eb-88ff-ecddc852e753.mp4

University Email verification on entering configure view

### CODE REVIEW

File: lib/core/node/common/university.dart [hunk 1]
```diff
@@ -35,6 +36,7 @@ class University extends TitleNode {
       : this.emblemImgURL = snapshot['emblemImgURL'],
         this.location = LatLng.fromJson(snapshot['location']),
         this.address = snapshot['address'],
+        this.domains = (snapshot['domains'] ?? []).cast<String>(),
```
Comment (jtjun) [inline lib/core/node/common/university.dart:39]: Wow,,,

``` dart
.cast<String>
```

File: lib/core/service/session.dart [hunk 2]
```diff
@@ -79,7 +86,7 @@ class Session {
         'ownerId': uid,
         'name': name,
         'userId': userId,
-        'isVerified': false,
+        'isVerified': true,
```
Comment (jtjun) [inline lib/core/service/session.dart:89]: nope, until user check email,
isVerified is false

File: lib/core/service/padong_auth.dart [hunk 3]
```diff
@@ -10,15 +10,25 @@
...
 import 'package:padong/core/shared/types.dart';
+import 'package:padong/core/shared/validator.dart';
 
 class PadongAuth {
   static final FirebaseAuth _auth = FirebaseAuth.instance;
 
   static Future<bool> isEmailVerified() async {
     if (_auth.currentUser == null) throw Exception('No Current User');
     await _auth.currentUser.reload();
-    return _auth.currentUser.emailVerified;
```
Comment (jtjun) [inline lib/core/service/padong_auth.dart:21]: `isEmailVerified` is check 'does user check his(her) email'.

**Review by jtjun — CHANGES_REQUESTED** [review]
Please check

File: lib/core/shared/types.dart [hunk 4]
```diff
@@ -141,6 +141,7 @@ enum SignUpResult {
   emailAlreadyInUse,
   IdAlreadyInUse,
   UniversityNotFound,
+  invalidUniversityEmail,
```
Comment (jtjun) [inline lib/core/shared/types.dart:144]: please unify with either camel case or pascal case.

``` dart
emailAlreadyInUse // camel
IdAlreadyInUse // pascal
UniversityNotFound // pascal
invalidUniversityEmail // camel
```

**Review by jtjun — CHANGES_REQUESTED** [review]
Please check

**Comment by VertexToEdge** [comment]
https://user-images.githubusercontent.com/37045096/114108351-e3c69900-990d-11eb-8136-d2eb6312b115.mp4

Change Email Test. and fix routing bug

**Pull request merged**